x/programs/rust/examples/single_nft/src/example.rs

```diff
@@ -0,0 +1,133 @@
+                    value: PROGRAM_ID.to_string(),
+                },
+                Param {
+                    param_type: ParamType::String,
+                    value: nft_name.to_string(),
+                },
+                Param {
+                    param_type: ParamType::U64,
+                    value: nft_name_length.to_string(),
+                },
+                Param {
+                    param_type: ParamType::String,
+                    value: nft_symbol.to_string(),
+                },
+                Param {
+                    param_type: ParamType::U64,
+                    value: nft_symbol_length.to_string(),
+                },
+                Param {
+                    param_type: ParamType::String,
+                    value: nft_uri.to_string(),
+                },
+                Param {
+                    param_type: ParamType::U64,
+                    value: nft_uri_length.to_string(),
+                },
+            ],
+            require: None,
+        },
+        Step {
+            endpoint: Endpoint::Key,
+            method: "create".to_string(),
+            max_units: 1000,
+            params: vec![
+                Param {
+                    param_type: ParamType::Id,
+                    value: PROGRAM_ID.to_string(),
+                },
+                Param {
+                    param_type: ParamType::Key(Key::Ed25519),
+                    value: "alice key".to_string(),
+                },
+            ],
+            require: None,
+        },
+        Step {
+            endpoint: Endpoint::Execute,
+            method: "mint_to".to_string(),
+            max_units: 1000,
+            params: vec![
+                Param {
+                    param_type: ParamType::Id,
+                    value: PROGRAM_ID.to_string(),
+                },
+                Param {
+                    param_type: ParamType::Key(Key::Ed25519),
+                    value: "alice_key".to_string(),
+                },
+            ],
+            require: None,
+        },
+    ];
+
+    Plan {
+        caller_key: "alice key".to_string(),
+        steps,
+    }
+}
+
+// TODO: remove when the simulator merges
+#[allow(dead_code)]
+fn run() -> Result<Vec<ExecuteResponse>, Box<dyn std::error::Error>> {
+    let nft_name = "MyNFT";
+    let binding = nft_name.len().to_string();
+    let nft_name_length: &str = binding.as_ref();
+
+    let nft_symbol = "MNFT";
+    let binding = nft_symbol.len().to_string();
+    let nft_symbol_length: &str = binding.as_ref();
+
+    let nft_uri = "ipfs://my-nft.jpg";
+    let binding = nft_uri.len().to_string();
+    let nft_uri_length: &str = binding.as_ref();
+
+    let plan = initialize_plan(
+        nft_name,
+        nft_name_length,
+        nft_symbol,
+        nft_symbol_length,
+        nft_uri,
+        nft_uri_length,
+    );
+    let client = Client::new("path to simulator".to_owned());
+
+    let tx = client.run::<ExecuteResponse>(&plan)?;
```

Comment:
This is still unnecessary
```suggestion
    let tx = client.run(&plan)?;
```

Reply:
^^